cpu/sam0_common/periph/adc.c

```diff
@@ -76,7 +76,8 @@ static void _adc_poweroff(void)
 
 static int _adc_configure(adc_res_t res)
 {
-    assert(res >= ADC_RES_8BIT && res <= ADC_RES_12BIT);
+    assert((res == ADC_RES_8BIT) || (res == ADC_RES_10BIT) ||
+           (res == ADC_RES_12BIT));
```

Comment:
I don't have a strong opinion on this but `assert(res <= ADC_RES_8BIT && res >= ADC_RES_12BIT);` should do the trick too, right? The confusing part for me then is that ADC_RES_12BIT < ADC_RES_8BIT, so a comment explaining that would be great.